metagpt/provider/azure_openai_api.py

```diff
@@ -43,8 +45,36 @@ def _make_client_kwargs(self) -> dict:
         return kwargs
 
     def _calc_usage(self, messages: list[dict], rsp: str) -> CompletionUsage:
-        return CompletionUsage(prompt_tokens=0, completion_tokens=0, total_tokens=0)
+        usage = CompletionUsage(prompt_tokens=0, completion_tokens=0, total_tokens=0)
+        if not self.config.calc_usage:
+            return usage
+
+        model_name = "gpt-35-turbo" if "gpt-3" in self.model.lower() else "gpt-4-turbo-preview"
+        try:
+            usage.prompt_tokens = count_message_tokens(messages, model_name)
+            usage.completion_tokens = count_string_tokens(rsp, model_name)
+        except Exception as e:
+            logger.error(f"usage calculation failed: {e}")
+
+        return usage
 
     @handle_exception
     def _update_costs(self, usage: CompletionUsage):
-        pass
+        if self.config.calc_usage and usage and self.cost_manager:
+            model_name = self._get_azure_model()
+            # More about pricing: https://azure.microsoft.com/en-us/pricing/details/cognitive-services/openai-service/
+            self.cost_manager.update_cost(usage.prompt_tokens, usage.completion_tokens, model_name)
+
+    def _get_azure_model(self) -> str:
```

Comment:
I think a simpler logic is needed than this function. The matching logic of this function lacks sufficient rationality and is an implicit condition

Reply:
Replaced by `pricing_plan` configuration